pkg/test/framework/components/istio/operator.go

```diff
@@ -739,7 +739,7 @@ func (i *operatorComponent) configureDirectAPIServerAccess(ctx resource.Context,
 
 func (i *operatorComponent) configureDirectAPIServiceAccessForCluster(ctx resource.Context, cfg Config,
 	c cluster.Cluster) error {
-	clusters := ctx.Clusters().Configs(c.Config())
+	clusters := ctx.AllClusters().Primaries(c)
```

Comment:
I think you need to make this configurable (secretsOnConfig = true/false) to do it one way or the other. I think you also need a similar change where the primary cluster secret is being created (around line 370).